pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -218,6 +223,28 @@ func (c AddOrUpdateChange) recordAppliedResource(savedRes ctlres.Resource) error
 		return fmt.Errorf("Calculating change after the save: %s", err)
 	}
 
+	// Use compact representation to take as little space as possible
+	// because annotation value max length is 262144 characters
+	// (https://github.com/vmware-tanzu/carvel-kapp/issues/48).
+	appliedResBytes, err := applyChange.AppliedResource().AsCompactBytes()
+	if err != nil {
+		return fmt.Errorf("Compact representation of applied resource: %s", err)
+	}
+
+	diff := applyChange.OpsDiff()
+
+	const resourceSizeMaxLen = 262144
+
+	// kapp deploy should work without adding disable annotation when resource size exceed
+	// (https://github.com/vmware-tanzu/carvel-kapp/issues/410)
+	if len(appliedResBytes) > resourceSizeMaxLen || len(diff.MinimalMD5()) > resourceSizeMaxLen {
+		if resourceWithHistoryDebug {
+			fmt.Printf("Unable to record resource, exceed limit of %d bytes (this can result in dumb diffing) \n",
```

Comment:
I do not think we wanna use the term "dumb" in our code
Maybe "might result in the diff showing deletion of fields applied by external agents"